src/api.d.ts

```diff
@@ -157,16 +157,14 @@ export interface AbstractAzExtTreeItem {
     isAncestorOfImpl?(contextValue: string | RegExp): boolean;
 }
 
-export type ResolvedAppResourceTreeItemBase = Partial<{ [P in keyof SealedAzExtTreeItem]: never }> & AbstractAzExtTreeItem;
+export type ResolvedAppResourceBase = Partial<{ [P in keyof SealedAzExtTreeItem]: never }> & Partial<AbstractAzExtTreeItem> & { contextValue?: never, contextValues?: string[] };
 
-export type ResolvedAppResource = ResolvedAppResourceTreeItemBase;
-
-export type ResolvedAppResourceTreeItem<T extends ResolvedAppResourceTreeItemBase> = AppResource & Omit<T, keyof ResolvedAppResourceTreeItemBase>;
+export type ResolvedAppResourceTreeItem<T extends ResolvedAppResourceBase> = AppResource & SealedAzExtTreeItem & Omit<T, keyof ResolvedAppResourceBase>;
 
 export type LocalResource = AzExtTreeItem;
 
 export interface AppResourceResolver {
-    resolveResource(subContext: ISubscriptionContext, resource: AppResource): vscode.ProviderResult<ResolvedAppResource>;
+    resolveResource(subContext: ISubscriptionContext, resource: AppResource): vscode.ProviderResult<ResolvedAppResourceBase>;
```

Comment:
Should we add `isApplicable` as something that must be defined as well? All resolvers probably need this, right?

Reply:
That's a good idea. A way to be more intelligent than simple type checking. It also solves our type/kind dilemma.

Reply:
I did these changes, though I called it `matchesResource` to align with some similar functions already on our APIs.